operatorcourier/validate.py

```diff
@@ -85,30 +93,27 @@ def _bundle_validation(self, bundle):
 
     def _crd_validation(self, bundleData):
         logger.info("Validating custom resource definitions.")
-        valid = True
-
         crds = bundleData[self.crdKey]
 
+        valid = True
         for crd in crds:
-            if "metadata" in crd:
-                if "name" in crd["metadata"]:
-                    logger.info("Evaluating crd %s", crd["metadata"]["name"])
-                else:
-                    self._log_error("crd metadata.name not defined.")
-                    valid = False
-            else:
-                self._log_error("crd metadata not defined.")
-                valid = False
-
-            if "apiVersion" not in crd:
-                self._log_error("crd apiVersion not defined.")
-                valid = False
+            for crd_required_nested_field in self.crd_required_nested_fields:
+                valid &= self._crd_fields_validation(crd_required_nested_field.split('.'),
```

Comment:
nit: Any benefit of using this oneliner over an if statement? It's already long enough to spread two lines

Reply:
Actually using an if statement will also make the line exceed 90 chars. I think the line is long mainly because the long and descriptive method names.

Reply:
Please be aware, that `&=` is a bitwise operator. But it seems this code intends to do a boolean operation, so probably

```python
valid = valid and crd_field_valid
```
would better express the intention here.

Reply:
Hi @csomh, thanks for the comment. I have updated this.